flatcar-tmpfiles: Copy over the shadow group and use it for shadow files

krnowak opened the pull request:
Some binaries acting on those files are now owned by the shadow group
with a sticky bit instead of having CAP_DAC_SEARCH (or
CAP_DAC_OVERRIDE) caps. For them to work, shadow and gshadow files
also need to be owned by the group shadow.

Will be tested together with the weekly updates.

krnowak committed:
flatcar-tmpfiles: Copy over the shadow group and use it for shadow files

Some binaries acting on those files are now owned by the shadow group
with a sticky bit instead of having CAP_DAC_SEARCH (or
CAP_DAC_OVERRIDE) caps. For them to work, shadow and gshadow files
also need to be owned by the group shadow.

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>

chewi reviewed (approved):
I don't think these ever used caps? They were previously SUID root. There is an old [Gentoo bug](https://bugs.gentoo.org/564896) open about using caps, so I'm a little surprised that floppym went with SGID recently instead, but I suppose non-caps support was needed too.

krnowak committed:
flatcar-tmpfiles: Avoid warnings when files are missing

In the logs there are warnings like:

cut: /sysroot/etc/passwd: No such file or directory
cut: /sysroot/etc/group: No such file or directory
cut: /sysroot/etc/shadow: No such file or directory
cut: /sysroot/etc/gshadow: No such file or directory

They are confusing - we are fine with those files being absent, but we
shouldn't be fine with printing such warnings. It misled me into
thinking of being an issue.

Also put the common part into a function, hopefully it's a bit easier
to read.

Signed-off-by: Krzesimir Nowak <knowak@microsoft.com>

krnowak commented:
I think these were SUID root for a long time, because of our modifications. Then they started using caps because I moved some package to portage-stable and then, recently, they became SGID shadow.

Also, I updated the PR for so flatcar-tmpfiles stops printing some warnings into journalctl. It initially confused me as being an issue.

krnowak commented:
Also changed chgrp to chown - there is no chgrp in initrd and this change was faster to make than modifying bootengine to include chgrp.

chewi commented on bin/flatcar-tmpfiles:
```diff
@@ -3,35 +3,59 @@
+        # It may fail if with /tmp.
+        tmp_file=$(mktemp --tmpdir="${ROOT}/etc")
+        cut --only-delimited --delimiter=":" --fields=1 "${root_file}" | \
+            sed -e 's/^/\^/' -e 's/$/:.*$/' >"${tmp_file}"
```
If you're using `--line-regexp`, then you don't need the anchors.

```suggestion
            sed -e 's/$/:.*/' >"${tmp_file}"
```

krnowak replied:
True. I copied was we used to have and expanded it to use long flag names. But didn't catch this one. Thanks.
